### PR TITLE
Better handling when switching to an account with disabled external links

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
To test:
Account A with external links
Account B w/o external links

Switch  between them and see that A always has links, whereas B never has them.

Thanks to @rullzer for pointing this out 🎉

### CODE REVIEW

File: src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java [hunk 1]
```diff
@@ -1335,53 +1335,50 @@ public void onAccountRemovedEvent(AccountRemovedEvent event) {
...
-                                runOnUiThread(() -> updateExternalLinksInDrawer());
+                            for (ExternalLink link : externalLinks) {
+                                externalLinksProvider.storeExternalLink(link);
                             }
-                        } else {
-                            sharedPreferences.edit().putInt(EXTERNAL_LINKS_COUNT, count + 1).apply();
                         }
                     } else {
-                        Log_OC.d("ExternalLinks", "links disabled");
+                        sharedPreferences.edit().putInt(EXTERNAL_LINKS_COUNT, count + 1).apply();
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java]: I might be mistaken, but don't we have a "PreferencesDao" with static methods to do read/write for preferences?

Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java]: We have a ArbitrayDataProvider now, but if I remember correctly this was introduced afterwards

Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java]: Ah, okay 👍

Comment (mario) [reply src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java]: This will at some point have to go into ArbitraryDataProvider.

**Comment by mario** [comment]
👍 

@AndyScherzinger ^_^

**Comment by AndyScherzinger** [comment]
👍 done :)

File: src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java (hunk 1 above)
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java]: merged, issue created for this: #1850

**Pull request merged**